app/src/main/java/com/sun/chat_04/ui/login/LoginActivity.kt

```diff
@@ -0,0 +1,153 @@
+package com.sun.chat_04.ui.login
+
+import android.content.Intent
+import android.support.v7.app.AppCompatActivity
+import android.os.Bundle
+import android.support.v4.content.ContextCompat
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.View
+import android.widget.Toast
+import com.facebook.CallbackManager
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.FirebaseDatabase
+import com.sun.chat_04.R
+import com.sun.chat_04.R.id
+import com.sun.chat_04.R.layout
+import com.sun.chat_04.data.remote.UserRemoteDataSource
+import com.sun.chat_04.data.repositories.UserRepository
+import com.sun.chat_04.ui.signup.SignUpActivity
+import com.sun.chat_04.util.Constants
+import kotlinx.android.synthetic.main.activity_login.buttonLoginEmailPass
+import kotlinx.android.synthetic.main.activity_login.buttonLoginFace
+import kotlinx.android.synthetic.main.activity_login.buttonLoginFb
+import kotlinx.android.synthetic.main.activity_login.editEmailLogin
+import kotlinx.android.synthetic.main.activity_login.editPassLogin
+import kotlinx.android.synthetic.main.activity_login.progressbarSignIn
+import kotlinx.android.synthetic.main.activity_login.textSignUp
+import java.lang.Exception
+
+class LoginActivity : AppCompatActivity(), View.OnClickListener
+    , LoginContract.View, TextWatcher {
+
+    private lateinit var presenter: LoginPresenter
```

Comment:
```suggestion
    private lateinit var presenter: LoginContract.Presenter
```
khai báo contract thì bản contract của em mới có giá trị